ga4gh/datamodel/variants.py

```diff
@@ -342,3 +343,118 @@ def getMetadata(self):
         # TODO: Implement this
         ret = []
         return ret
+
+
+def _encode_value(value):
+    if isinstance(value, (list, tuple)):
+        return [str(v) for v in value]
+    else:
+        return [str(value)]
+
+
+_nothing = object()
+
+
+def is_empty_iter(it):
+    """Return True iff the iterator is empty or exhausted"""
+    return next(it, _nothing) is _nothing
+
+
+class HtslibVariantSet(VariantSet):
+    """
+    Class representing a single variant set backed by a directory of indexed
+    VCF or BCF files.
+    """
+
+    def __init__(self, variantSetId, vcfPath, use_bcf=True):
+        self._variantSetId = variantSetId
+        self._created = protocol.convertDatetime(datetime.datetime.now())
+        self._chromFileMap = {}
+
+        for pattern in ['*.bcf', '*.vcf.gz']:
+            for filename in glob.glob(os.path.join(vcfPath, pattern)):
+                self._addFile(filename)
+
+    def _addFile(self, filename):
+        varFile = pysam.VariantFile(filename)
+        if not varFile.index:
```

Comment:
We're using explicit comparison with None elsewhere for this sort of thing, i.e.
```python
if varFile.index is None:
    # do stuff
```